### PR TITLE
FIX: Granting staff status should auto-approve users waiting approval

### PR DESCRIPTION
**Pull request by romanrizzi**

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -471,7 +471,6 @@ def invite_admin
     end
 
     user.active = true
-    user.approved = true
```
Comment (SamSaffron) [inline app/controllers/admin/users_controller.rb]: do we need an extra test somewhere to confirm that if there is no reviewable the user still gets approved?

Comment (romanrizzi) [reply app/controllers/admin/users_controller.rb]: This is already covered by the `users_controller_spec`. We could also add an additional test for the user class.

File: app/models/concerns/roleable.rb [hunk 2]
```diff
@@ -48,4 +50,15 @@ def set_permission(permission_name, value)
     save_and_refresh_staff_groups!
   end
 
+  private
+
+  def auto_approve_user
+    reviewable = ReviewableUser.where(status: Reviewable.statuses[:pending]).find_by(target: self)
```
Comment (eviltrout) [inline app/models/concerns/roleable.rb]:
```suggestion
    reviewable = ReviewableUser.find_by(target: self, status: Reviewable.statuses[:pending])
```
This is equivalent I think and slightly shorter.

Comment (eviltrout) [reply app/models/concerns/roleable.rb]: You could also do `if reviewable = ...` and cut out a line.

**Pull request merged**